website/docs/d/all_namespaces.html.markdown

```diff
@@ -0,0 +1,29 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_all_namespaces"
+sidebar_current: "docs-kubernetes-data-source-all-namespaces"
+description: |-
+  Lists all namespaces within a cluster.
+---
+
+# kubernetes_all_namespaces
+
+This data source provides a mechanism for listing the names of all available namespaces in a Kubernetes cluster.
+It can be used check for existence of a specific namespaces or to apply another resource to all or a subset of existing namespaces in a cluster.
+
+In Kubernetes, namespaces provide a scope for names and are intended as a way to divide cluster resources between multiple users.
+
+## Example Usage
+
+```hcl
+data "kubernetes_all_namespaces" "allns" {}
+
+output "all-ns" {
+  value = data.kubernetes_all_namespaces.allns.namespaces
+}
+
+output "ns-present" {
+  value = contains(data.kubernetes_all_namespaces.allns.namespaces, "kube-system")
+}
+
+```
```

Comment:
Datasources usually have an attribute reference section:
````suggestion
```

## Attribute Reference

The following attributes are exported:

* `namespaces` - A list of existing namespace names.
````

See:
- https://github.com/terraform-providers/terraform-provider-kubernetes/blame/v1.10.0/website/docs/d/secret.html.markdown#L51-L54
- https://github.com/terraform-providers/terraform-provider-google/blame/v2.20.0/website/docs/d/datasource_compute_region_instance_group.html.markdown#L58-L69
- https://github.com/terraform-providers/terraform-provider-null/blame/v2.1.2/website/docs/data_source.html.markdown#L55-L64